src/plugins/intel_cpu/src/nodes/istft.cpp

```diff
@@ -0,0 +1,111 @@
+    }
+
+    const auto istft_op = as_type_ptr<op::v16::ISTFT>(op);
+
+    m_is_frame_size_const = is_type<op::v0::Constant>(istft_op->get_input_node_ptr(FRAME_SIZE_IDX));
+    m_is_frame_step_const = is_type<op::v0::Constant>(istft_op->get_input_node_ptr(FRAME_STEP_IDX));
+    if (istft_op->get_input_size() == 5) {
+        m_is_signal_length_const = is_type<op::v0::Constant>(istft_op->get_input_node_ptr(SIGNAL_LENGTH_IDX));
+    }
+    m_center = istft_op->get_center();
+    m_normalized = istft_op->get_normalized();
+}
+
+void ISTFT::getSupportedDescriptors() {
+    const auto input_size = getParentEdges().size();
+    if (input_size < 4 || input_size > 5) {
+        THROW_CPU_NODE_ERR("ISTFT has incorrect number of input edges.");
+    }
+    if (getChildEdges().empty()) {
+        THROW_CPU_NODE_ERR("ISTFT has incorrect number of output edges.");
+    }
+}
+
+void ISTFT::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty()) {
+        return;
+    }
+
+    auto dataPrecision = getOriginalInputPrecisionAtPort(DATA_IDX);
+    if (!one_of(dataPrecision, ov::element::f32)) {
+        dataPrecision = ov::element::f32;
+    }
+
+    std::vector<PortConfigurator> configurators({{LayoutType::ncsp, dataPrecision},
+                                                 {LayoutType::ncsp, dataPrecision},
+                                                 {LayoutType::ncsp, ov::element::i32},
+                                                 {LayoutType::ncsp, ov::element::i32}});
+    if (getOriginalInputsNumber() == 5) {
+        configurators.emplace_back(LayoutType::ncsp, ov::element::i32);
+    }
+
+    addSupportedPrimDesc(configurators, {{LayoutType::ncsp, dataPrecision}}, impl_desc_type::ref_any);
+}
+
+bool ISTFT::needPrepareParams() const {
+    return false;
+}
+
+bool ISTFT::created() const {
+    return getType() == Type::ISTFT;
+}
+
+void ISTFT::execute(const dnnl::stream& strm) {
+    const auto signal_length =
+        getOriginalInputsNumber() == 5 ? (getSrcDataAtPortAs<const int32_t>(SIGNAL_LENGTH_IDX))[0] : -1;
+    ov::reference::istft(getSrcDataAtPortAs<const float>(DATA_IDX),
+                         getSrcDataAtPortAs<const float>(WINDOW_IDX),
+                         getDstDataAtPortAs<float>(0),
+                         ov::Shape{getSrcMemoryAtPort(DATA_IDX)->getStaticDims()},
+                         ov::Shape{getSrcMemoryAtPort(WINDOW_IDX)->getStaticDims()},
+                         (getSrcDataAtPortAs<const int32_t>(FRAME_SIZE_IDX))[0],
+                         (getSrcDataAtPortAs<const int32_t>(FRAME_STEP_IDX))[0],
+                         signal_length,
+                         m_center,
+                         m_normalized);
+}
+
+void ISTFT::executeDynamicImpl(const dnnl::stream& strm) {
+    execute(strm);
+}
+
+bool ISTFT::needShapeInfer() const {
+    return !m_is_signal_length_const || (m_is_frame_size_const && m_is_frame_step_const) || Node::needShapeInfer();
```

Comment:
I see few possible issues with this needShapeInfer:
1. `signal_length` is optional and defaults to `false`, so if not provided this will always return `true`
2. `(m_is_frame_size_const && m_is_frame_step_const)` seem like should be negated (if any is not const - return `true`)
3. Based on ISTFT shape_inference, as long as `signal_length` is const, then `frame_size` and `frame_step` are not part of output shape computation, shouldn't they be ignored?

Reply:
Good point, fixed to check if the signal length input is provided, and added negation.